### PR TITLE
[GH-#512] Transaction pool sync

### LINKED ISSUE #512: Transaction pool sync
Pool sync was changed in epoch at some point, we have to implement this to make sync work since we would receive messages we can't handle

### PR DESCRIPTION
**Pull request by d-velev**
Comparison of pools between epoch nodes is done with MPTs so for now we tell the epoch node that we have an empty tree in order to get its transactions (this means that syncing isn't optimized, I'll try to fix this after the 31st). There might be some messy stuff leftover which was needed to match the genesis hash, I'll clean that up soon.

closes #512

### CODE REVIEW

File: apps/aecore/lib/aecore/chain/block.ex [hunk 1]
```diff
@@ -24,11 +24,39 @@ defmodule Aecore.Chain.Block do
     @version
   end
 
+  # @spec genesis_header() :: Header.t()
+  # defp genesis_header do
+  #   header = Application.get_env(:aecore, :pow)[:genesis_header]
+  #   struct(Header, header)
+  # end
+
+  def genesis_header do
```
Comment (gspasov) [inline apps/aecore/lib/aecore/chain/block.ex]: Why create this function if there is one already doing this in the [Genesis module](https://github.com/aeternity/elixir-node/blob/master/apps/aecore/lib/aecore/chain/genesis.ex#L71)?

**Comment by thepiwo** [comment]
what is the status here?

File: apps/aecore/lib/aecore/chain/block.ex (hunk 1 above)
Comment (d-velev) [reply apps/aecore/lib/aecore/chain/block.ex]: As I said there is some stuff that needs to be removed, I'll take care of it

Comment (gspasov) [reply apps/aecore/lib/aecore/chain/block.ex]: I didn't know you are still working on this and aware of the issue in hand. I guess there was some miscommunication on my part.

**Pull request merged**